sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/logging/JsonLoggingConfigurationRunListener.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.logging;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import org.springframework.boot.SpringApplicationRunListener;
+
+/**
+ * Configures Spring Boot to log as JSON based on the SDA standard setting {@code
+ * ENABLE_JSON_LOGGING}. This is implemented as {@link SpringApplicationRunListener} to cover as
+ * many log statements as possible - including logs that are printed before bean processing is
+ * completed. However, there are still a few logs of the early initialization phase that can't be
+ * covered.
+ */
+public class JsonLoggingConfigurationRunListener implements BootstrapRegistryInitializer {
+
+  private static final String ENABLE_JSON_LOGGING_ENV_NAME = "ENABLE_JSON_LOGGING";
+  private static final String ENABLE_JSON_LOGGING_PROPERTY_NAME = "enable.json.logging";
+  private static final String TRUE = "true";
+  private static final String LOGGING_CONFIG_PROPERTY_NAME = "logging.config";
+  private static final String JSON_LOGGING_CONFIG_RESOURCE =
+      "classpath:org/sdase/commons/spring/boot/web/logging/logback-json.xml";
+
+  @Override
+  public void initialize(BootstrapRegistry registry) {
+    if (isJsonLoggingEnabled()) {
+      configureJsonLogging();
+    }
+  }
+
+  private void configureJsonLogging() {
+    System.setProperty(LOGGING_CONFIG_PROPERTY_NAME, JSON_LOGGING_CONFIG_RESOURCE);
+  }
+
+  private boolean isJsonLoggingEnabled() {
+    return isJsonLoggingEnabledByEnv() || isJsonLoggingEnabledByProperty();
```

Comment:
Is there a way to use Spring injection to retrieve the value? E.g. by `@Value`? I'm curious if it would try to access both resources automatically.

Reply:
No, because the log must be configured before the environment is resolved. You can follow that in `SpringApplication` when you dig into and check where listeners are called. Though I did not find the exact location where logging is configured, it is before the first „hook“ after environment is resolved.

Reply:
`@Value` is not available at all in this phase because run listeners and bootstrap initializers are not beans. They are created before the context. At a later phase some run listeners get programmatic access to the environment, but that phase is too late for log configuration.